### PR TITLE
Add hide/show channel

### PR DESCRIPTION
**Pull request by bevzaanton**
# Submit a pull request

## Description of the pull request
- add API methods to `Client` to show/hide a channel
- add methods to `Channel` to show/hide itself
- add methods for end-client to show/hide channel. After success request, the channel will be removed from `ChannelListViewModel.channels`

## Methods for end-client:
**ChannelListViewModel**
`public void hideChannel(@NotNull Channel channel)` - hides the channel from queryChannels for the user until a message is added and remove from the current list of channels
`public void showChannel(@NotNull Channel channel)` - removes the hidden status for a channel and remove from the current list of channels

### CODE REVIEW

File: library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelListViewModel.java [hunk 1]
```diff
@@ -151,6 +155,50 @@ public void setChannelFilter(FilterObject filter) {
         this.filter = filter;
     }
 
+    /**
+     * hides the channel from queryChannels for the user until a message is added and remove from the current list of channels
+     *
+     * @param channel the channel needs to hide
+     */
+    public void hideChannel(@NotNull Channel channel) {
```
Comment (tbarbugli) [inline library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelListViewModel.java]: probably good to add a callback handler in the signature to allow handling success / failure in the view

File: library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java [hunk 2]
```diff
@@ -566,6 +568,64 @@ public void onFailure(Call call, Throwable t) {
...
     }
 
+    /**
+     * hides the channel from queryChannels for the user until a message is added
+     *
+     * @param channel  the channel needs to hide
+     * @param callback the result callback
+     */
+    public void hideChannel(@NonNull Channel channel, @NotNull ShowHideChannelCallback callback) {
+        mService.hideChannel(channel.getType(), channel.getId(), apiKey, clientID,
```
[the remark this answers is not in the document]
Comment (tbarbugli) [reply library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java]: we need to add a TODO here

// TODO: track hidden state in `Room`

File: library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java [hunk 3]
```diff
@@ -566,6 +568,64 @@ public void onFailure(Call call, Throwable t) {
...
+     * @param channel  the channel needs to show
+     * @param callback the result callback
+     */
+    public void showChannel(@NonNull Channel channel, @NotNull ShowHideChannelCallback callback) {
+        mService.showChannel(channel.getType(), channel.getId(), apiKey, clientID,
+                new HideChannelRequest(user.getId()))
+                .enqueue(new Callback<ShowHideChannelResponse>() {
+                    @Override
+                    public void onResponse(Call<ShowHideChannelResponse> call, Response<ShowHideChannelResponse> response) {
+                        if (response.isSuccessful()) {
```
Comment (tbarbugli) [inline library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java]: I don't think we need to check if response is successful, that's already done. Please have a look at the `RetrofitClient` class, I might be wrong about this.

File: library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelListViewModel.java [hunk 4]
```diff
@@ -151,6 +155,50 @@ public void setChannelFilter(FilterObject filter) {
...
+            }
+        });
+    }
+
+    /**
+     * removes the hidden status for a channel and remove from the current list of channels
+     *
+     * @param channel the channel needs to show
+     */
+    public void showChannel(@NotNull Channel channel) {
```
Comment (tbarbugli) [inline library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelListViewModel.java]: see my remarks on hideChannel

File: library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java [hunk 5]
```diff
@@ -566,6 +568,64 @@ public void onFailure(Call call, Throwable t) {
...
 
+    /**
+     * hides the channel from queryChannels for the user until a message is added
+     *
+     * @param channel  the channel needs to hide
+     * @param callback the result callback
+     */
+    public void hideChannel(@NonNull Channel channel, @NotNull ShowHideChannelCallback callback) {
+        mService.hideChannel(channel.getType(), channel.getId(), apiKey, clientID,
+                new HideChannelRequest(user.getId()))
```
[the remark this answers is not in the document]
Comment (bevzaanton) [reply library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java]: If I don't pass userId I receive error form API:
`{"code":4,"message":"HideChannel failed with error: \"invalid json data\"","StatusCode":400,"duration":"0.00ms"}`

**Pull request merged**